### PR TITLE
Build Nuget package for ulib

### PR DESCRIPTION
**Pull request by kant2002**
Whole idea is to have ulib to be distributed, so somebody can use F* with F#
without specifying paths to ulib.dll

### CODE REVIEW

**Comment by kant2002** [comment]
Not sure if this should supersede https://github.com/FStarLang/FStar/pull/2308, but I feel that it contains a lot of things which may be split and modified/createdindependently.

- FStar SDK (collection of tools for running making integration easier with .NET) That's `targets` files. For now this is only extraction file, but maybe something more, I do not think thought that.
- FStar runtime (that's actual fstar.exe + z3.exe + binary dependencies)
- ULib itself

I understand that right now FStar.exe and ulib.dll is very closely tied together, but not sure that this is requirement. From what I see Ulib provide couple supplementary functions (which maybe not even used by F# extraction) and older Fstar compiler can work perfectly with newer Ulib.

I see modern F* + F# workflow like following

```
dotnet new -i Fstarlang
dotnet new fstar.library
```

which create approximately following fsproj
```
<Project Sdk="FStarLang.Sdk">
  <PropertyGroup>
    <TargetFramework>net6.0</TargetFramework>
  </PropertyGroup>
  <ItemGroup>
    <Compile Include="Program.fst" />
  </ItemGroup>
  <ItemGroup>
    <PackageReference Update="FStar.Ulib" Version="1.0.0" />
    <PackageReference Update="FSharp.Core" Version="4.3.4" />
  </ItemGroup>
</Project>
```

which hopefully just work, and you can work within Visual Studio or Visual Studio Code.

On the `dotnet pack` vs `nuspec` file. I usually prefer store all Nuget metadata in project file, since they are autoamatically applied to assembly as well, so there no need to mess with version in the Nuspec too much. 
For example `dotnet pack -c Release /p:VersionSuffix=2022.06.06` will produce version `1.0.0-2022.06.06` and all properties would be captured from project file which serves as substitute for `nuspec`.

@cc mateuszbujalski

**Comment by mateuszbujalski** [comment]
Looks like a right way forward to me now that fstar has been migrated to sdk project and newer framework. 
I think #2308 should be scrapped now, but you might want to reuse parts of it (e.g. the script to publish nugets to nuget.org - it was discussed with @tahina-pro at the time so perhaps a good base). 

Regarding the fstar.exe and ulib coupling, I'm fairly sure fstar.exe won't work without ulib **sources**. The ulib.dll is only relevant for F#, and can be shipped separately.

**Comment by lucasteles** [comment]
> see modern F* + F# workflow like following

looks great

**Comment by kant2002** [comment]
Would be good to receive some feedback from @nikswamy or @tahina-pro

File: ulib/fs/VS/Makefile [hunk 1]
```diff
@@ -1,20 +1,21 @@
...
+	$(DOTNET) build UlibFS.sln
+	$(DOTNET) pack ../ulibfs.fsproj -o ../../../nuget/
 
 install-packages:
-	$(MSBUILD) /t:Restore UlibFS.sln
+	$(DOTNET) restore $(DOTNET_PARAMS) UlibFS.sln
 
 clean:
-	$(MSBUILD) /t:clean UlibFS.sln
+	$(MSBUILD) clean $(DOTNET_PARAMS) UlibFS.sln
```
Comment (nikswamy) [inline ulib/fs/VS/Makefile:21]: Should this be $(DOTNET)?

Comment (kant2002) [reply ulib/fs/VS/Makefile:21]: Umm. Yes. Sigh.

**Comment by kant2002** [comment]
Hi  @nikswamy or @tahina-pro . Would like to nudge you to review this PR.

**Pull request merged**